core/trino-main/src/main/java/io/trino/execution/scheduler/faulttolerant/EventDrivenFaultTolerantQueryScheduler.java

```diff
@@ -2369,14 +2370,10 @@ private void recordFinishStats()
         {
             long finishTime = System.currentTimeMillis();
```

Comment:
This is wrong. currentTimeMillis is using a system clock that can go backwards. To measure latencies nanoTime should be used (monotonically increasing)

Reply:
Does not matter for this use - but I will change

Reply:
Sure. It was a general comment. I've seen some user reported errors that duration was negative